src/nlp_expr.jl

```diff
@@ -502,6 +489,8 @@ end
 
 moi_function(x::Number) = x
 
+moi_function(x::AbstractArray) = moi_function.(x)
```

Comment:
Okay, one edge-case.

What is the `moi_function` of `norm(::Vector{VariableRef})`. Is it a `Vector{MOI.VariableIndex}` or is it a `MOI.VectorOfVariables`?

```julia
julia> op_norm = NonlinearOperator(LinearAlgebra.norm, :norm)
NonlinearOperator(norm, :norm)

julia> model = Model()
A JuMP Model
Feasibility problem with:
Variables: 0
Model mode: AUTOMATIC
CachingOptimizer state: NO_OPTIMIZER
Solver name: No optimizer attached.

julia> @variable(model, x[1:3])
3-element Vector{VariableRef}:
 x[1]
 x[2]
 x[3]

julia> op_norm(x)
norm(VariableRef[x[1], x[2], x[3]])

julia> moi_function(op_norm(x))
norm(┌                    ┐
│MOI.VariableIndex(1)│
│MOI.VariableIndex(2)│
│MOI.VariableIndex(3)│
└                    ┘)
```

Reply:
And this method

Reply:
@blegat thoughts?